### PR TITLE
Funnels simple mode

### LINKED ISSUE #6776: Proposal - Funnels simple mode
## Proposal
Introduce a simple mode for funnels, and lower prominence of advanced features.

## Context
With all the changes and improvements we did to funnels, we ended up with a product that can be complex to understand for new users, particularly non-technical and/or new to product analytics.

## Approach
Implement the [following proposal](https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=4989%3A28013) (screenshots below, **very much open to feedback**) as an A/B test.

**Simple mode (default)**
image

**Advanced mode**
image

**Simple mode when advanced customization has been applied**
image

## Goal
Proposed approach should increase Discoveries on funnels (segmenting particularly on new vs. existing users). In addition, we need to make sure only an acceptable drop on usage for advanced features is seen. 

Looking for feedback @marcushyett-ph @kpthatsme @clarkus @mariusandra @Twixes

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

**🎯The goal of this PR is increasing funnel Discoveries, particularly in new users.** We'll run an A/B test for this.

- Closes #6776. Implements a "simple mode" for funnels in which advanced options are grouped and hidden. The rationale is that a simpler interface will incentivize more funnel creation, and advanced users will find advanced options anyways. 
- Updates general styling to match latest designs by @clarkus.
- Updates style of internal test users filter.
- Removes legacy unused styles.

**Postgres version**

**Horizontal version - collapsed - no advanced options applied**

**Vertical version - collapsed - no advanced options applied**

**Vertical version - expanded**

**Vertical version - collapsed - advanced options applied**

## How did you test this code?
- All permutations of the following configurations: Postgres vs. Clickhouse; `funnel-simple-mode` on vs. off, `5730-funnel-horizontal-ui` on vs. off, groups functionality enabled vs. disabled.

### CODE REVIEW

File: frontend/src/scenes/insights/AggregationSelect.tsx [hunk 1]
```diff
@@ -8,9 +8,10 @@ const UNIQUE_USERS = -1
 interface AggregationSelectProps {
     aggregationGroupTypeIndex: number | undefined
     onChange: (aggregationGroupTypeIndex: number | undefined) => void
+    style?: React.CSSProperties
 }
 
-export function AggregationSelect({ aggregationGroupTypeIndex, onChange }: AggregationSelectProps): JSX.Element {
+export function AggregationSelect({ aggregationGroupTypeIndex, onChange, style }: AggregationSelectProps): JSX.Element {
```
Comment (alexkim205) [inline frontend/src/scenes/insights/AggregationSelect.tsx:14]: Nit: can we use a classname or even better a boolean property here?

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx [hunk 2]
```diff
@@ -26,8 +26,14 @@ import { insightLogic } from 'scenes/insights/insightLogic'
 import { AggregationSelect } from 'scenes/insights/AggregationSelect'
 import { groupsModel } from '~/models/groupsModel'
 import { TaxonomicFilterGroupType } from 'lib/components/TaxonomicFilter/types'
+import { FunnelTabWithSimpleMode } from './FunnelTabWithSimpleMode'
 
 export function FunnelTab(): JSX.Element {
+    const { featureFlags } = useValues(featureFlagLogic)
+    return featureFlags[FEATURE_FLAGS.FUNNEL_SIMPLE_MODE] ? <FunnelTabWithSimpleMode /> : <FunnelTabOld />
```
Comment (alexkim205) [inline frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:33]: Nice I love that you separated this for easy cleanup

**Review by alexkim205 — CHANGES_REQUESTED** [review]
Wow I love this new updated look and the separation of simple/advanced options. It makes it so much easier to get started with funnels. 

Some feedback

- It looks like the stylistic changes are also grouped in with the `funnel-simple-mode` feature. If it's not too much of a hassle I would prefer the style update to be separated out so that but up to you to decide :) 
- Opening and closing advanced options will trigger another loadResults call. Usually unnoticeable but definitely slows things down when queries are expensive.
- It feels like there should be an option to reset advanced options, especially when their "dirtiness" is exposed in a "X options applied tag"
- Some style things

Screen Shot 2021-11-22 at 10 42 49 AM

- Nit: removing '(s)'s in aggregation dropdown

Screen Shot 2021-11-22 at 10 43 20 AM

**Review by clarkus — CHANGES_REQUESTED** [review]
image
- While the filter labeling is super obvious, I do think it biases filters over the actual graph series parameters. Can we use a badge instead?

- The aggregation option for non-postgres feels over-sized. Can we instead size it to the character width of the currently selected option? If not that, then the character width of the most verbose option?
- We previously had descriptions inline to help describe each section of the query options. Are those being removed or changed purposefully? 
- I feel like the conversion window could be more prominent or possibly not an advanced option. It feels as critical as aggregation method to me. 
- We call the section "advanced options" and we summarize its state with "1 setting applied". I don't know that we need to name what's applied specifically or that it's applied. Really we need to summarize things that are different than the default values. A badge or some other simple count should be enough to call out that something is applied.

**Comment by paolodamico** [comment]
Thanks for the review @alexkim205 & @clarkus, ready for another look. Some pushback/clarification on a few items, everything else is addressed.

Re @alexkim205 
- _"Nit: removing '(s)'s in aggregation dropdown"_
    - Unrelated to this PR, worth checking with @macobo as this may affect other parts of the app.
- Opted for keeping the stylistic changes here for now. To avoid duplication of work, if we move forward in a couple of weeks, it won't matter anymore, and if we don't we can then rollback these changes and import the relevant style changes.

----------

Re @clarkus 
- _"The aggregation option for non-postgres feels over-sized."_
    - No changes were introduced in this PR, but agree with the premise. Given that it affects multiple insights, would you mind opening a separate PR, pls?
- "We previously had descriptions inline to help describe each section of the query options. Are those being removed or changed purposefully?"   
    - No changes to that introduced in this PR. We have those descriptions behind the info icon. Happy to change it if it makes sense.

----------

**Multiple adjustments**


**Simplified badge**

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 3]
```diff
@@ -1317,6 +1344,28 @@ export const funnelLogic = kea<funnelLogicType<openPersonsModelProps>>({
             }
         },
     }),
+    urlToAction: ({ actions }) => ({
```
Comment (alexkim205) [inline frontend/src/scenes/funnels/funnelLogic.ts:1347]: If we keep `funnel_advanced` in filters and don't persist it into a separate value, we might be able to get rid of a lot of boilerplate here.

**Review by clarkus — APPROVED** [review]

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 4]
```diff
@@ -1317,6 +1344,28 @@ export const funnelLogic = kea<funnelLogicType<openPersonsModelProps>>({
...
+        '/insights': (
+            _,
+            __,
+            {
+                funnel_advanced,
+            }: {
+                funnel_advanced?: boolean
+            }
+        ) => {
+            if (funnel_advanced) {
```
Comment (alexkim205) [inline frontend/src/scenes/funnels/funnelLogic.ts:1357]: If none of the advanced options are on (aka advancedOptionsUsedCount === 0) we should set advanced mode to false.

**Comment by alexkim205** [comment]
Overall looking pretty good! I removed some boilerplate in favor of moving `funnel_advanced` into filters and added a logic test for not triggering a load result on toggling that menu. There're still a few style bugs that already existed before this PR, so I'll leave that up to your discretion.

Screen Shot 2021-11-24 at 10 16 47 AM

Screen Shot 2021-11-24 at 10 39 31 AM

**Review by alexkim205 — APPROVED** [review]

**Pull request merged**